inc/namespace.php

```diff
@@ -196,3 +198,77 @@ function enqueue_scripts() {
 	// Print queued scripts.
 	print_head_scripts();
 }
+
+/**
+ * Schedule index maintenance daily.
+ */
+if( ! wp_next_scheduled( __NAMESPACE__ . '\\index_maintenance' ) ) {
+	wp_schedule_event( time(), 'daily',  __NAMESPACE__ . '\\index_maintenance' );
+}
+
+
+/**
+ * Delete old Analytics indexes in Elasticsearch.
+ */
+function delete_old_indexes() : ?array {
+	// Index age in days.
+	$duration = apply_filters( 'altis.analytics.index_age', 7 );
+
+	// If duration has been set out of the 60-day limit, default to 7 days and warn user.
+	if ( $duration < 1 || $duration > 60 ) {
+		$duration = 7;
+		trigger_error( 
+			'Analytics data retention period must be between 1 and 60 days, defaulting to 7.', 
+			E_USER_WARNING 
+		);
+	}
+
+	// Get index name by date. 
+	$date = new \DateTime();
```

Comment:
Rather than `\DateTime()` it helps PHP optimise itself a bit to add `use DateTime;` at the top of the file then you don't need the backslash here. Not a blocker really but it's consistent with our coding standards.